gulpfile.js/tasks/changelog.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
+
+var gulp = require('gulp')
+var gutil = require('gulp-util')
+var proc = require('child_process')
+
+var runCommand = function (cmd) {
+  return new Promise(function (resolve, reject) {
+    proc.exec(cmd, function (err, stdout, stderr) {
+      if (err) {
+        reject(err)
+      } else if (stderr) {
+        reject(stderr)
+      } else {
+        resolve(stdout)
+      }
+    })
+  })
+}
+
+var getCurrentCommit = function (commit) {
+  return new Promise(function (resolve, reject) {
+    if (commit) {
+      resolve(commit)
+    } else {
+      var cmd = 'git rev-parse HEAD'
+      runCommand(cmd)
+        .then(function (commit) {
+          resolve(commit)
+        })
+    }
+  })
+}
+
+var getChangedFiles = function (commit) {
+  if (!commit) {
+    return Promise.reject(new Error('No commit given'))
+  }
+
+  var cmd = 'git diff --name-only master ' + commit
+  return runCommand(cmd)
+}
+
+var checkForChangelog = function (files) {
+  if (!files.includes('CHANGELOG.md')) {
```

Comment:
Potentially a preference thing but I would ask why are you using `Promise.reject` and `Promise.resolve` rather than just returning a `return new Promise(function (resolve, reject){ ...`? It feels like there is a mix of both ways in this file. IMO it would be good to choose one or the other? Feels like a shorthand too far rather than defining this as a `Promise` at the function/interface level. Thoughts?